python/pyspark/sql/udf.py

```diff
@@ -42,10 +42,15 @@ def _create_udf(f, returnType, evalType):
                     PythonEvalType.SQL_GROUPED_AGG_PANDAS_UDF):
 
         import inspect
+        import sys
         from pyspark.sql.utils import require_minimum_pyarrow_version
 
         require_minimum_pyarrow_version()
-        argspec = inspect.getargspec(f)
+
+        if sys.version_info[0] < 3:
+            argspec = inspect.getargspec(f)
+        else:
+            argspec = inspect.getfullargspec(f)
```

Comment:
Shall we add a small comment while we are here like `'getargspec ' is deprecated since version 3.0 and calling it with type hints causes an actual issue. See SPARK-23569`?

Reply:
can do.